### ISSUE TITLE
Error while installing tweasel-cli

### ISSUE THREAD

**Issue by baltpeter** [tweaselORG/autopy#7]
```
root@ryzen ~ # npm i -g tweasel-cli
npm WARN deprecated querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
npm ERR! code 1
npm ERR! path /usr/lib/node_modules/tweasel-cli/node_modules/cyanoacrylate
npm ERR! command failed
npm ERR! command sh -c node scripts/postinstall.js
npm ERR! node:internal/process/promises:288
npm ERR!             triggerUncaughtException(err, true /* fromPromise */);
npm ERR!             ^
npm ERR! 
npm ERR! [Error: EEXIST: file already exists, symlink '2to3-3.11' -> '/root/.cache/autopy/python/3.11.3/bin/2to3'] {
npm ERR!   errno: -17,
npm ERR!   code: 'EEXIST',
npm ERR!   syscall: 'symlink',
npm ERR!   path: '2to3-3.11',
npm ERR!   dest: '/root/.cache/autopy/python/3.11.3/bin/2to3'
npm ERR! }
npm ERR! 
npm ERR! Node.js v18.16.0

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2023-06-08T00_44_37_110Z-debug-0.log
```

**Comment by baltpeter** [comment]
I worked after I manually installed CA first, followed by CLI. Random thought: Is npm maybe parallelizing postinstall scripts? Do we need a lockfile?

**Comment by baltpeter** [comment]
Seems like that is not the problem. I got the same error when running `tweasel record-traffic`.

**Comment by baltpeter** [comment]
Actually, it does seem to be a race after all. I can't reproduce this when I just run a simple autopy script. But if I run two scripts that both just run `downloadPython()` at the same time, I get the same problem.

**Comment by baltpeter** [comment]
Indeed, if I add logging to `tweasel record-traffic`, I see that there are two calls to `downloadPython`.

**Comment by baltpeter** [comment]
Ooooh… In appstraction, both `android.ts` and `ios.ts` do `const venv = getVenv(venvOptions);` at the top-level.

**Comment by baltpeter** [comment]
But I'm pretty sure that that isn't the issue with the postinstall script. So, I'm guessing that the ones from appstraction and CA are in fact run in parallel.

**Comment by baltpeter** [comment]
I really don't want to have to implement my own lockfile mechanism. There is just way too much that can go wrong. https://github.com/moxystudio/node-proper-lockfile seems like a decent choice.

**Comment by baltpeter** [comment]
For now, I'll go for overzealous locking rather than trying to optimize this all the way. We can follow up later.

**Comment by baltpeter** [comment]
@zner0L Can you have a look at this when you have to verify my fix (not now)? I've left the issue open and the [release](https://github.com/tweaselORG/autopy/releases/tag/v1.1.1) is currently marked as a prerelease, since I really wasn't too confident in my way too deep into the night fix. But from the testing we've done so far, it seems to have held up at least.

**Comment by zner0L** [comment]
I looked at it and deemed it fine. Thank you!
